scripts/tests/test_signature.py

```diff
@@ -0,0 +1,70 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+from scripts.signature import bias_signatures
+from pathlib import Path
+import json
+
+
+def read_from_file(file_path):
+    return Path(file_path).read_text()
+
+
+def test_bias(base_signatures, contender_signatures):
+    return bias_signatures(
+        json.loads(base_signatures), json.loads(contender_signatures), 10
+    )
+
+
+class SignatureTest(unittest.TestCase):
+    def test_bias(self):
+        # Remove a signature
+        _, return_value = test_bias(
+            """{"reverse": ["(array(T)) -> array(T)"]}""",
+            """{"reverse": []}""",
```

Comment:
So this script is supposed to test compatibility, plus return the bias for new signatures? Wouldn't it be better to do this as different steps, somehow?

For example, assuming that I need to make a legit change to the signatures; say I'm fixing a function that was mistakenly registered before. It would be nice to be able to get an error regarding the incompatible change (which you could choose to skip), but still having the biased fuzzer run.

Reply:
If you have an idea of how we can make a step optional and still warn the user about breaking changes let me know , I will modify it, but at the moment I am not sure how we can make step warn user about breaking changes and still go through without it being missed.